springfox-spi/src/main/java/springfox/documentation/spi/service/ProjectionProviderPlugin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ *  Copyright 2017 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+
+package springfox.documentation.spi.service;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+
+import org.springframework.plugin.core.Plugin;
+
+import com.fasterxml.classmate.ResolvedType;
+import com.google.common.base.Optional;
+
+import springfox.documentation.spi.DocumentationType;
+
+public interface ProjectionProviderPlugin extends Plugin<DocumentationType> {
```

Comment:
I feel like this plugin shouldn have something other than`DocumentationType` as the discriminator. It should be something like `ProjectionType` which can have one of the following types
- Jackson
- Spring-Data-Rest
- Something we haven't thought off.

Then we can select the right plugin rather than pick the first one the pluginManager finds.Thoughts?

Reply:
For that we should have some explicit configuration. Any way we won't have all plugins at one time. So, Jackson JsonView Plugin has the lowest priority. If Spring-Data-Rest is included in classpath it will overwrites with its own plugin with higher priority. And custom plugin has the highest priority will overwrite all plugins.